conduit/pipeline/pipeline.go

```diff
@@ -394,6 +408,13 @@ func (p *pipelineImpl) Start() {
 		for {
 		pipelineRun:
 			metrics.PipelineRetryCount.Observe(float64(retry))
+			if uint64(retry) >= p.cfg.RetryCount {
```

Comment:
nit: if p.cfg.RetryCount is uint64, retry can be set to uint64 type at line 403 to get rid of type cast here.